packages/web-app-files/src/mixins/actions/restore.ts

```diff
@@ -29,6 +29,12 @@ export default {
               return false
             }
 
+            const { manager, editor } = this.space?.spaceRoles
+            const isProjectSpace = isProjectSpaceResource(this.space)
+            if (isProjectSpace && ![...manager, ...editor].includes(this.user.uuid)) {
```

Comment:
Could you introduce two functions `isManager` and `isEditor` on the project space interface and implement them with the code you have here? That would remove some duplication... could be used as follows:
```js
if (isProjectSpaceResource(this.space) && (this.space.isManager(this.user.uuid) || this.space.isEditor(this.user.uuid))
```